validator/lib/validateBoundingVolume.js

```diff
@@ -0,0 +1,123 @@
+'use strict';
+
+var Promise = require('bluebird');
+var Cesium = require('cesium');
+var defined = Cesium.defined;
+var Cartesian3 = Cesium.Cartesian3;
+var Rectangle = Cesium.Rectangle;
+var Cartographic = Cesium.Cartographic;
+
+module.exports = validateVolume;
+
+function validateVolume(tileset) {
+    return new Promise(function(resolve) {
+        validateNode(tileset.root, tileset, resolve);
+    });
+}
+
+var scratchCartographic = new Cartographic();
+var scratchContentCartesian = new Cartesian3();
+var scratchTileCartesian = new Cartesian3();
+var scratchContentRectangle = new Rectangle();
+var scratchTileRectangle = new Rectangle();
+
+function regionInsideRegion(contentRegion, tileRegion) {
+    var contentRectangle = Rectangle.unpack(contentRegion, 0, scratchContentRectangle);
+    var tileRectangle = Rectangle.unpack(tileRegion, 0, scratchTileRectangle);
+    var maxContentHeight = contentRegion[5];
+    var minContentHeight = contentRegion[4];
+    var maxTileHeight = tileRegion[5];
+    var minTileHeight = tileRegion[4];
+    return (Rectangle.contains(tileRectangle,  Rectangle.northwest(contentRectangle, scratchCartographic))
+        && Rectangle.contains(tileRectangle, Rectangle.southwest(contentRectangle, scratchCartographic))
+        && Rectangle.contains(tileRectangle, Rectangle.northeast(contentRectangle, scratchCartographic))
+        && Rectangle.contains(tileRectangle, Rectangle.southeast(contentRectangle, scratchCartographic)))
+        && (maxContentHeight < maxTileHeight)
+        && (minContentHeight > minTileHeight);
+}
+
+function sphereInsideSphere(contentSphere, tileSphere) {
+    var contentRadius = contentSphere[3];
+    var tileRadius = tileSphere[3];
+    var contentCenter = Cartesian3.unpack(contentSphere, 0, scratchContentCartesian);
+    var tileCenter = Cartesian3.unpack(tileSphere, 0, scratchTileCartesian);
+    var distance = Cartesian3.distance(contentCenter, tileCenter);
+    return distance  <= (tileRadius - contentRadius);
+}
+
+function sphereInsideRegion(contentSphere, tileRegion) {
+    var contentRadius = contentSphere[3];
+    Cartesian3.unpack(contentSphere, 0, scratchContentCartesian);
+    Cartographic.fromCartesian(scratchContentCartesian, Cesium.Ellipsoid.WGS84, scratchCartographic);
+    var tileRectangle = Rectangle.unpack(tileRegion, 0, scratchTileRectangle);
+    var width = tileRegion[2] - tileRegion[0];
+    var height = tileRegion[3] - tileRegion[1];
+    return ((Rectangle.contains(tileRectangle, scratchCartographic)) &&
+            (2*contentRadius <= width && 2*contentRadius <= height));
+}
```

Comment:
There are a couple of problems with this test, it doesn't take the tile region's min and max height into account and doesn't account for when the sphere is near the edge of the region.

I haven't fully thought this through, but if you get the normals of each side of the region, you can use those to get 6 points on the sphere. Then check that each point is inside the region. If may be helpful to write a `cartographicInsideRegion` function used here and in `regionInsideRegion`.

This function may be helpful for finding the normals: https://github.com/AnalyticalGraphicsInc/cesium/blob/3d-tiles/Source/Scene/TileBoundingRegion.js#L160